packages/styleguide/src/components/ExpandableLink/ExpandableLink.tsx

```diff
@@ -0,0 +1,148 @@
+  return typeof document === 'object'
+    ? ReactDOM.createPortal(children, document.body)
+    : null
+}
+
+const ExpandableLink = ({
+  children,
+  attributes,
+  title,
+  description,
+  href,
+}: Props) => {
+  const [colorScheme] = useColorContext()
+  const [expandedLink, setExpandedLink] = useState<StateProps>(undefined)
+  const timeOutRef = useRef<NodeJS.Timeout>(null)
+
+  const toggleLinkInfoBox = (position: CSSProperties) => {
+    if (expandedLink) {
+      setExpandedLink(undefined)
+    } else {
+      setExpandedLink({ title, description, href, position: position })
+    }
+  }
+
+  const getCalloutPosition = (event) => {
+    const target = event.target as HTMLAnchorElement
+    const targetHeight = target.offsetHeight
+    const targetBottom = target.getBoundingClientRect().bottom
+    const targetTop = target.getBoundingClientRect().top
+    const targetMiddle =
+      target.getBoundingClientRect().top + target.offsetHeight / 2
+    const isMultiline =
+      parseInt(
+        window.getComputedStyle(target).getPropertyValue('line-height'),
+      ) <= targetHeight
+
+    const hoveringOnTopLine = event.clientY <= targetMiddle
+    const displayAbove = event.clientY >= window.innerHeight / 2
+    const displayToLeft = event.clientX >= window.innerWidth / 2
+
+    return {
+      // if callout renders BELOW target AND
+      // if the target link wraps on multiple lines AND
+      // the client is hovering on the TOP line of the multi-line wrapped link THEN
+      // use the target middle as baseline
+      top: displayAbove
+        ? 'auto'
+        : isMultiline && hoveringOnTopLine
+        ? targetMiddle + MARGIN
+        : targetBottom + MARGIN,
+      // if callout renders ABOVE the target link AND
+      // if the target link wraps on multiple lines AND
+      // the client is hovering on the BOTTOM line of the multi-line wrapped link THEN
+      // use the target middle as baseline
+      bottom: displayAbove
+        ? isMultiline && !hoveringOnTopLine
+          ? window.innerHeight - targetMiddle + MARGIN
+          : window.innerHeight - targetTop + MARGIN
+        : 'auto',
+      left: displayToLeft ? 'auto' : event.clientX,
+      right: displayToLeft ? window.innerWidth - event.clientX : 'auto',
+    }
+  }
+
+  const showInfoBox = (event) => {
+    clearTimeout(timeOutRef.current)
+    const position = getCalloutPosition(event)
+    timeOutRef.current = setTimeout(() => {
+      toggleLinkInfoBox(position)
+    }, DELAY)
+  }
+
+  const removeInfoBox = () => {
+    clearTimeout(timeOutRef.current)
+    timeOutRef.current = setTimeout(() => setExpandedLink(undefined), DELAY)
+  }
+
+  return (
+    <a
+      {...attributes}
+      {...styles.link}
+      {...colorScheme.set('color', 'text')}
+      {...colorScheme.set('textDecorationColor', 'textSoft')}
+      aria-label={description}
+      onClick={(event) => {
+        event.preventDefault()
+        showInfoBox(event)
+      }}
```

Comment:
Shouldn't we allow cmd click to directly get to the source?
```suggestion
      href={href}
      onClick={(event) => {
        if (shouldIgnoreClick(event)) {
          return
        }
        event.preventDefault()
        showInfoBox(event)
      }}
```
(`shouldIgnoreClick` function would need to be moved to styleguide)

Reply:
done